server/groundstation.py

```diff
@@ -90,11 +92,37 @@ def uav_thread(self):
             time.sleep(0.1)
 
     def image_thread(self):
-        while True:
-            run = self.image.retreive_images()
-            if run:
-                self.logger.debug("[Image] %s", run)
-            time.sleep(0.1)
+        if self.config["uav"]["images"]["connection"]:  # Initialize a socket connection
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                s.bind((self.config["uav"]["images"]["host"], self.config["uav"]["images"]["port"]))
+                s.listen()
+                conn, addr = s.accept()
+                with conn:
+                    self.logger.warning("[Image] Socket connection established with " + addr)
+                    curr_image = ""
+                    while True:
+                        resp = conn.recv(1024)
+                        if not resp:
+                            break
+                        resp = resp.decode("utf-8")
+                        loc = resp.find(" ")
+                        if loc != -1:
+                            curr_image += resp[:loc]
+                            res = self.image.process_image(curr_image)
+                            curr_image = resp[loc + 1:]
+                            self.logger.debug("[Image] Successfully retreived image from socket")
+                            if res:
+                                self.logger.info("[Image] Successfully identified ODLC from Image")
+                            conn.sendall(bytes("Image Complete", "utf-8"))
+                        else:
+                            curr_image += resp
+                    self.logger.warning("[Image] Lost socket connection with " + addr)
+        else:  # Use a dummy connection
+            while True:
+                run = self.image.retreive_images()
+                if run:
+                    self.logger.info("[Image] Successfully identified ODLC from Image")
+                time.sleep(0.1)
 
     def async_calls(self):
         self.interop_telem_thread = Thread(target=self.telemetry_thread)
```

Comment:
You should add the image loop thread calls into here.

Reply:
It is here... lines 136-138

Reply:
Oh...

I skimmed over it and I didn't see it lol